CMakeLists.txt

```diff
@@ -5,6 +5,10 @@ if(CMAKE_VERSION VERSION_EQUAL 3.0 OR CMAKE_VERSION VERSION_GREATER 3.0)
     cmake_policy(SET CMP0042 NEW)
 endif()
 
+if(POLICY CMP0058)
+  cmake_policy(SET CMP0058 NEW)
```

Comment:
Also, I think cmake ninja was able to detect some of our custom build commands generate byproducts and implicitly set BYPRODUCTS.